low-level/actuators/impl/generic/realstor_encl.py

```diff
@@ -111,31 +130,45 @@ def perform_request(self, jsonMsg):
                             component_type,
                             resource)
                 else:
-                    logger.error("Unsupported system request :{}".format(component_type))
+                    logger.error("Unsupported system request :{}".format(enclosure_request))
 
         except Exception as e:
             logger.exception("Error while getting details for JSON: {}".format(jsonMsg))
             response = {"Error": e}
 
         return response
 
-    def make_response(self, component_details, component, component_type, resource_id):
+    def make_response(self, component_details, component, component_type,
+            resource_id, ctrl_action = None):
 
         resource_type = "enclosure:{}:{}".format(component, component_type)
         epoch_time = str(int(time.time()))
         alert_id = mon_utils.get_alert_id(epoch_time)
+        alert_type = "GET"
+        severity = "informational"
+        ctrl_action_lst = ['shutdown', 'restart']
 
-        if component == "fru":
+        if component == "fru" and not ctrl_action:

```

Comment:
dont we have request type available, which can be used here, like GET or control action

Reply:
Existing code will run on the condition of component type ("fru"). and the controller shutdown actuator component name is also "fru". that's why I didn't change the existing check like **component == "fru"**. we can differentiate request type using ctrl_action ('shutdown', 'restart'). so I kept here.
>dont we have request type available, which can be used here, like GET or control action

Yes. we have but now we have three different alert_type ('GET', 'control:restart', 'control:shutdown').
that will increase the code of line. I feel that written code will better.

Reply:
Made change